### PR TITLE
feat(rabbit)[OK-284] Add RabbitMQ microservice for handling mailing system with proxy client

### PR DESCRIPTION
**Pull request by marcepanowyy**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -51,3 +65,13 @@ services:
...
+
+  rabbitmq:
+    image: rabbitmq:3.12.2-management
+    ports:
+      - "${RABBITMQ_PORT}:${RABBITMQ_PORT}"
+      - "${RABBITMQ_MANAGEMENT_PORT}:${RABBITMQ_MANAGEMENT_PORT}"
+    container_name: obelisk_rabbitmq_container
+    environment:
+      - RABBITMQ_USER=${RABBITMQ_USER}
+      - RABBITMQ_PASSWORD=${RABBITMQ_PASSWORD}
```
Comment (Wajktor13) [inline docker-compose.yml]:
```suggestion
      - RABBITMQ_USER
      - RABBITMQ_PASSWORD
```

File: docker-compose.yml [hunk 2]
```diff
@@ -27,21 +26,36 @@ services:
...
       - GOOGLE_CLIENT_SECRET
       - GOOGLE_CALLBACK_URL
 
+      - MAILING_HOST
+      - MAILING_PORT
+      - MAILING_SECURE
+      - MAILING_USER
+      - MAILING_PASS
+
   mongo:
```
Comment (Wajktor13) [inline docker-compose.yml]:
```suggestion
    environment:
      - SERVER_HOST
      - SERVER_PORT
      - SOCKET_GW_PORT
      - RABBITMQ_HOST
      - RABBITMQ_PORT
      - RABBITMQ_QUEUE
      - DB_NAME
      - DB_USER
      - DB_PASSWORD
      - DB_HOST=mongo
      - DB_PORT
      - JWT_SECRET
      - JWT_EXPIRES_IN
      - GOOGLE_CLIENT_ID
      - GOOGLE_CLIENT_SECRET
      - GOOGLE_CALLBACK_URL
      - MAILING_HOST
      - MAILING_PORT
      - MAILING_SECURE
      - MAILING_USER
      - MAILING_PASS

  mongo:
```

File: server/src/config/mogo.config.ts
Comment (Wajktor13) [inline server/src/config/mogo.config.ts:1]: wrong name

**Review by Wajktor13 — CHANGES_REQUESTED** [review]

File: server/src/config/mogo.config.ts
Comment (marcepanowyy) [reply server/src/config/mogo.config.ts:1]: fixed

File: docker-compose.yml (hunk 2 above)
Comment (marcepanowyy) [reply docker-compose.yml]: fixed

File: docker-compose.yml (hunk 1 above)
Comment (marcepanowyy) [reply docker-compose.yml]: fixed

**Review by Wajktor13 — DISMISSED** [review]

**Review by Wajktor13 — APPROVED** [review]
This PR:

1. Dockerizes RabbitMQ, adds new envs for rabbit
2. Adds configs (mongo.config.ts, logger.config.ts, …)
3. Adds mailing module
4. Integrates mailing module with rest of the serve
5. Moves stuff between shared/interfaces/decorators
6. … something else?

There should be at least one separate PR for each of this points. Combining it all into one results in a huge cognitive load and makes the PR unreadable and impossible to review correctly.

**Pull request merged**